### DISCUSSION TITLE
Ability to export the created React / Next.js app

### DISCUSSION THREAD

**Discussion by iron3oxide** [lowdefy/lowdefy#1307] [Ideas and feature requests]
Let me preface this by saying that this is a long shot and by no means do I expect this to be possible in the near or even at any point in the future.

From what I understand, lowdefy works by converting the yaml / json files into React components, thereby creating a React (or Next.js for v4) app and then serving that app via the lowdefy server.

Would it be possible to have a CLI option to export, meaning to stop after building and pasting the generated files to a specified location, ideally even in some kind of developer-friendly format? That way one could also use a lowdefy frontend in something like a Tauri or Electron app (I think). 

This feels like a naive question you have probably dismissed some time ago, so if you did, please document the reason here, I couldn't find an answer elsewhere.

**Comment by Gervwyk** [comment]
HI @iron3oxide 

It depends on what you mean by export, but v4 is simply a next.js app. so once built you should be able to do anything with it that you can be with a next.js app. The nice part is that the Lowdefy build runs before the next.js build, so you have access to the unbuilt files like if you would have coded it from scratch - check out the `.lowdefy/dev` folder in a v4 app.

**Reply by iron3oxide** [reply]
Wow, that's quite the game changer, now I'm even more excited for v4! Your wording seems to imply that there are some v4 apps already. Where can I find them?

**Reply by Gervwyk** [reply]
We're already running some of our in-house prod apps on v4 alpha to test it in the wild, it's doing well 🦥 
Currently the team is busy with docs and dev-ex to wrap it up, maybe one or two more features (can never have enough of those). 
You can give it a spin by running `npx lowdefy@4.0.0-alpha.22 init` or dev.. 

There might be a few more breaking changes in the pipeline but you can view the initial (most basic / unofficial) migration guide here: https://github.com/lowdefy/lowdefy/blob/main/packages/docs/howto/v4Migration/v4-migration.md

**Reply by Gervwyk** [reply]
There is also this discussion on plugins (which is really the game changer imo): https://github.com/lowdefy/lowdefy/discussions/1251
Also follow the progression on pre-releases here: https://github.com/lowdefy/lowdefy/releases

**Reply by iron3oxide** [reply]
I guess it depends on perspective, to me this means I can write simple, declarative yaml and still enjoy practically all the benefits of a javaScript frontend while not being locked in by some subscription or even having to pay a single cent (remind me to sponsor you once that's possible). That's huge for me and other people without frontend experience and exactly what I've been looking for for years.

**Reply by Gervwyk** [reply]
Exactly! Our goal is to empower all devs to get useful stuff on the web with ease - everyone can think of something that can live in the browser but don't care much for css, etc.. But also, and for the growth of the project, monetization is on the roadmap but we plan to have it open-core, but with some paid nice-to-have services.

**Reply by iron3oxide** [reply]
Well, as long as that core is at least comparable to what the project offers now, I'll be the last to complain. Might I suggest the [mkdocs-material sponsorware strategy](https://squidfunk.github.io/mkdocs-material/insiders/)? You could have a private repo for sponsors with nice-to-have features and release some of them for each sponsorship goal that has been hit.
